contracts/DecentralizedKV.sol

```diff
@@ -104,7 +102,8 @@ contract DecentralizedKV {
             lastKvIdx = lastKvIdx + 1;
         }
         paddr.kvSize = uint24(data.length);
-        paddr.hash = bytes24(keccak256(data));
+        uint256 nChunkBits = generateChunkBits(data.length);
+        paddr.hash = bytes24(MerkleLib.merkleRoot(data, CHUNK_SIZE, nChunkBits));
```

Comment:
Could we have an optimized `merkleRootWithMinTree()` so that it just calculates the root of the data with minimal tree size (instead of doing extra calculations?)

Reply:
Could you please elaborate?
Shouldn't a standard Merkle tree spanned by a length of data.length / CHUNK_SIZE ?

Reply:
`merkleRoot()` currently asked for the size of the tree as the tree can be arbitrarily bigger as long as its leafs cover all the data leaves.

E.g., a data with 5 leaves will need a tree with minimal of 8 leaves, but the trees will 16, 32, 64 leaves also work.

Reply:
So the `nChunkBits` control how big a tree can be, right?
Right now, the calculation is, for example,
Data.length divided by CHUNK_SIZE, gets a chunk number, say 5, we will round up next number which is of 2^n, 8, then the bitslen is 3. 
As long as we control `nChunkBits`, we control the Merkle tree spanning.

Reply:
Yes, but what I mean is that we could optimize the calculation in `merkleRootWithMinTree()` which returns the root hash of the minimal tree without `nChunkBits` as a parameter.

Reply:
Okay I had an offline talk with Qi. Now I know the request is to simplify the API........
It should be quite intuitive.